### ISSUE TITLE
Plaid ML V.0.7 doesn't increase trainning accuracy

### ISSUE THREAD

**Issue by mchuv** [plaidml/plaidml#1066]
Hi , I have a simple CNN network with only 5 CNN layers , if I train it with PlaidML V.0.6.2 the training accuracy increases with every iteration , but with the same network wit PlaidML V.0.7 the accuracy doesn't increase it varies between 0.4 and 0.7 , I have waited 5 hours to see if the accuracy increases (I'm using the same training and validation images).

Another thing is that V.0.7 is slower than V.0.6.2:

With V.0.6.2 :
-------------

3458/3458 [==============================] - 584s 169ms/step - loss: 3.8358 - acc: 0.1146 - val_loss: 3.9880 - val_acc: 0.1121
Epoch 2/30
3458/3458 [==============================] - 581s 168ms/step - loss: 3.7588 - acc: 0.1246 - val_loss: 3.8405 - val_acc: 0.1403
Epoch 3/30

With V.0.7 :
----------

Epoch 1/30
3458/3458 [==============================] - 663s 192ms/step - loss: 4.5332 - acc: 0.0452 - val_loss: 4.5464 - val_acc: 0.0549
Epoch 2/30
3458/3458 [==============================] - 661s 191ms/step - loss: 4.3479 - acc: 0.0616 - val_loss: 4.5695 - val_acc: 0.0518

I'm working with MacBook Pro 2019 15", with OS X Catalina 10.15.4, with Python 3.7.6. ()

This is the summary of the Keras Model :

Layer (type)                 Output Shape              Param #   
=================================================================
conv2d_1 (Conv2D)            (None, 96, 96, 16)        160       
_________________________________________________________________
max_pooling2d_1 (MaxPooling2 (None, 48, 48, 16)        0         
_________________________________________________________________
batch_normalization_1 (Batch (None, 48, 48, 16)        64        
_________________________________________________________________
conv2d_2 (Conv2D)            (None, 48, 48, 64)        9280      
_________________________________________________________________
max_pooling2d_2 (MaxPooling2 (None, 24, 24, 64)        0         
_________________________________________________________________
batch_normalization_2 (Batch (None, 24, 24, 64)        256       
_________________________________________________________________
conv2d_3 (Conv2D)            (None, 24, 24, 128)       73856     
_________________________________________________________________
max_pooling2d_3 (MaxPooling2 (None, 12, 12, 128)       0         
_________________________________________________________________
batch_normalization_3 (Batch (None, 12, 12, 128)       512       
_________________________________________________________________
conv2d_4 (Conv2D)            (None, 12, 12, 64)        8256      
_________________________________________________________________
max_pooling2d_4 (MaxPooling2 (None, 6, 6, 64)          0         
_________________________________________________________________
batch_normalization_4 (Batch (None, 6, 6, 64)          256       
_________________________________________________________________
conv2d_5 (Conv2D)            (None, 6, 6, 8)           520       
_________________________________________________________________
batch_normalization_5 (Batch (None, 6, 6, 8)           32        
_________________________________________________________________
flatten_1 (Flatten)          (None, 288)               0         
_________________________________________________________________
dropout_1 (Dropout)          (None, 288)               0         
_________________________________________________________________
dense_1 (Dense)              (None, 128)               36992     
_________________________________________________________________
dropout_2 (Dropout)          (None, 128)               0         
_________________________________________________________________
dense_2 (Dense)              (None, 133)               17157     
=================================================================
Total params: 147,341
Trainable params: 146,781
Non-trainable params: 560

The diference or special case in this network is that the con2d_4 and conv2d_5 uses a 1x1 filter to finally reduce the number of trainable parameters.

**Comment by agj60** [comment]
Similar problem: mnist very simple model doesn't converge on GPU, but does on CPU.

**Comment by ufosoftwarellc** [comment]
I have a similar problem with 0.6.4.  The accuracy improves using Colab and on my machine using the CPU but it does not increase when using Plaid ML.  My requirements.txt file is as follows:
scikit-learn
scikit-image
matplotlib
seaborn
ipython
jupyter
scipy
numpy
pandas
pillow
pydot
coremltools==3.0
plaidml==0.6.4
plaidbench==0.6.4
keras==2.2.4
tensorflow==1.14

My imports are as follows
import os
os.environ["KERAS_BACKEND"] = "plaidml.keras.backend"
import keras
import numpy as np
from keras.preprocessing.image import ImageDataGenerator 
from keras.models import Sequential
from keras.layers import *
from keras import optimizers, callbacks
%matplotlib inline
import matplotlib.pyplot as plt

There must be an issue with my environment or my imports.

**Comment by sector84** [comment]
Similar problem: train custom cnn with a few layers and accuracy doesn't converge on GPU, but does on CPU.
My configuration
- MacBook Pro (15-inch, 2018 Radeon Pro 560X)
- OS X Catalina 10.15.6
- Python 3.8
- PlaidML 0.7.0

I did some research and found out that without BatchNormalization layers it works correctly, but ones these layers added the accuracy stops

The model I try to train on cifar-10 dataset
```
        model = Sequential()
        inputShape = (height, width, depth)
        model.add(Conv2D(32, (3, 3), padding="same",input_shape=inputShape))
        model.add(Activation("relu"))
        model.add(BatchNormalization())
        model.add(Conv2D(32, (3, 3), padding="same"))
        model.add(Activation("relu"))
        model.add(BatchNormalization())
        model.add(MaxPooling2D(pool_size=(2, 2)))
        model.add(Dropout(0.25))

        model.add(Conv2D(64, (3, 3), padding="same"))
        model.add(Activation("relu"))
        model.add(BatchNormalization())
        model.add(Conv2D(64, (3, 3), padding="same"))
        model.add(Activation("relu"))
        model.add(BatchNormalization())
        model.add(MaxPooling2D(pool_size=(2, 2)))
        model.add(Dropout(0.25))

        model.add(Flatten())
        model.add(Dense(512))
        model.add(Activation("relu"))
        model.add(BatchNormalization())
        model.add(Dropout(0.5))

        # softmax classifier
        model.add(Dense(classes))
        model.add(Activation("softmax"))
```
python minivggnet_cifar10.py --output output/cifar10_minivggnet_with_bn.png
Using plaidml.keras.backend backend.
[INFO] loading CIFAR-10 data...
[INFO] compiling model...
INFO:plaidml:Opening device "metal_amd_radeon_pro_560x.0"
[INFO] training network...
Train on 50000 samples, validate on 10000 samples
Epoch 1/40
50000/50000 [=====] - 139s 3ms/step - loss: 2.5165 - acc: 0.1010 - val_loss: 5357647833738091.0000 - val_acc: 0.1000
Epoch 2/40
50000/50000 [=====] - 137s 3ms/step - loss: 2.4248 - acc: 0.0992 - val_loss: 205880806582150.7500 - val_acc: 0.1000


If I remove any BatchNormalization layers the train process works properly
python minivggnet_cifar10.py --output output/cifar10_minivggnet_without_bn.png
Using plaidml.keras.backend backend.
[INFO] loading CIFAR-10 data...
[INFO] compiling model...
INFO:plaidml:Opening device "metal_amd_radeon_pro_560x.0"
[INFO] training network...
Train on 50000 samples, validate on 10000 samples
Epoch 1/40
50000/50000 [==============================] - 88s 2ms/step - loss: 1.7988 - acc: 0.3402 - val_loss: 1.5421 - val_acc: 0.4489
Epoch 2/40
50000/50000 [==============================] - 87s 2ms/step - loss: 1.3605 - acc: 0.5084 - val_loss: 1.1453 - val_acc: 0.5879
Epoch 3/40

**Comment by cortictechnology** [comment]
I have observed the same problem, with BatchNormalization layer in the model, the validation accuracy never increased, and is substantially lower than that with out BatchNormalization layer. Tested on 0.7.0 as well. This affects a lot of common usages.
